server/routes/api/chart.py

```diff
@@ -97,6 +97,22 @@ def keep_latest_data(timeseries):
     max_date = max(timeseries)
     return { max_date: timeseries[max_date] }
 
+    
+def keep_necessary_data(timeseries, dates):
```

Comment:
Is there a more accurate word to use? Necessary is very broad, perhaps specified?